### DISCUSSION TITLE
Save Distribution Plot Not Working

### DISCUSSION THREAD

**Discussion by nbungart1999** [phoebe-project/phoebe2#469] [Questions]
When I am trying to save my distribution plot it is not actually saving anything but it still runs. This is the plot line:  _ = b.plot_distribution_collection('ball_around_optimized_solution', show=True, save='dist_plot.pdf'). The plot still runs and shows it in the notebook but it does not actually save the plot. The save feature does still work fine for the normal b.plot() though.

**Comment by kecnry** [comment]
I filed a bug report #470, but probably won't be able to get to this for a few weeks.  In the meantime, it should return the figure object and you should be able to save directly from matplotlib.  If you can't get that working, please let me know!

**Reply by nbungart1999** [reply]
Ok thanks! Yeah we're running it in wsl and it wasn't returning anything so I just wanted to see if there was a work around but now we know it's a bug

**Reply by kecnry** [reply]
If you import matplotlib (`import matplotlib.pyplot as plt`) before plotting, I think that `plt.savefig('dist_plot.pdf')` should work.  Let me know - if it doesn't, I'll try to prioritize working on the bug fix so you're not left waiting too long (or there's always screenshots ;-) ).

**Reply by nbungart1999** [reply]
Ok, I will but how do I use the plt since I was plotting with phoebe?

**Reply by kecnry** [reply]
The following works for me:

```
import phoebe
b=phoebe.default_binary()
b.add_distribution('teff', phoebe.gaussian_around(200), allow_multiple_matches=True)

import matplotlib.pyplot as plt
b.plot_distribution_collection()
plt.savefig('test.png')
```

**Reply by nbungart1999** [reply]
Ok I'll try that!

**Reply by nbungart1999** [reply]
That method does work for now!
